### ISSUE TITLE
Project status / Taking contributions?

### ISSUE THREAD

**Issue by bitwit** [s1ddok/Fiber2D#12]
Hi, I'm pretty interested in this project and was wondering what the status is on development/whether the project is at a point where you want contributions and have an idea where help is most useful

It's great to see this effort out there. I really am not sure where to turn to get a cross-platform engine written in swift that has an API like cocos2d

**Comment by s1ddok** [comment]
@bitwit the main problem now is the ease of use. this is what stops most people from using it. The project is mostly in alpha-stage, but core functionality works well. Once we can create straightforward build process (probably only when Swift PM will support Build Settings) then we could start extending features. 

Current implementation has some bugs that need to be fixed, but that's on a lower priority.

**Comment by bitwit** [comment]
@s1ddok Thanks very much for your response and time spent on this project. I'm really enthusiastic about the opportunities Fabric can bring and would love to contribute more. This seems like it could be a great successor to cocos2d-iphone. 
I've got an early-days project in SpriteKit right now that I'd love to keep platform agnostic while remaining written in Swift, even if it means starting having to get deeper into some rendering code.

**Comment by s1ddok** [comment]
@bitwit well, Fiber2D operates on the same abstractiob level as SpriteKit, so I wouldn’t say it has more rendering code, unless you want to do something fancy

**Comment by bitwit** [comment]
Sorry, to clarify I meant things like `SKSpriteNode`, `SKTileMapNode` and `SKLabelNode`. It is my understanding Fiber2D has not yet expanded on these kinds of area.

**Comment by s1ddok** [comment]
@bitwit well, Fiber2D does support sprite nodes, but is truly lacking labels and tilemaps, which are on roadmap for sure

**Comment by bitwit** [comment]
@s1ddok I've had a little more time to investigate in the last few days. Got the Mac stuff running easily but iOS I've gotten a little stuck with `Cbgfx` module not being found. Will put this in another ticket though.
I see what you are saying, the build process right now is a little cumbersome which stops people from using it, though this issue seems quite temporary and fixable through even just a little more scripting. 

Another thing that is probably a bigger blocker for SpriteKit users like myself right now though, is portability of the code we've already written. I think where my interest lies in this project right now is working on a companion module that mimics the SpriteKit APIs as closely as possible (no .sks file support) but just wraps Fiber2D -- called something like `OpenSpriteKit` or `SprightKit`, idk 😂
I think that getting the necessary support to make a module like this in place would naturally bring some contributions back to the Fiber2D library while working toward a partial-migration plan for existing SpriteKit users who want to get on this cross-platform wagon.

Thanks again for your time and thoughts. I look forward to spending more time investigating over the holidays.

**Comment by norman784** [comment]
Indeed, I think every game swift engine need like you said @bitwit a companion, abstraction layer between existing sprite kit code and fiber2d, but to accomplish that the framework need to mimic the behaviors in the same way (don't used really sprite kit so don't know the api, neither fiber2d). But I'm kinda looking for something like this.

**Comment by Anywhere-Music-Player** [comment]
@s1ddok 
Many large companies create their iOS applications for natively from iPhone using a Swift, but they also need to port some of them to andirod, and here's what happens:

https://blog.readdle.com/why-we-use-swift-for-android-db449feeacaf

Any comments, ideas about this? 

I'm just working with cocos2d-x and C++, and this like has no future for mobile games.. too much time just for use C++, but with Swift this should be dramatically changed... as I believe. 
What do you think?

**Comment by Lancelotbronner** [comment]
Hello, I was wondering if there was interest by the authors to keep this project going?

**Comment by s1ddok** [comment]
@Lancelotbronner hey! thanks for your interest in the project. due to the many reasons it is now in paused stage, even though I still believe it can be used as a foundation for a full-fledged cross platform 2d game engine. 

I personally cannot maintain the project anymore but I'll be glad to review any contributions
